sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -85,8 +87,9 @@ class JacksonParser(
   private def makeStructRootConverter(st: StructType): JsonParser => Iterable[InternalRow] = {
     val elementConverter = makeConverter(st)
     val fieldConverters = st.map(_.dataType).map(makeConverter).toArray
+    val jsonFilters = new JsonFilters(filters, st)
```

Comment:
No big deal but wouldn't it better to just use `NoopFilters` when the switch is off? For example,

```scala
val jsonFilters = if (SQLConf.get.jsonFilterPushDown) {
  new JsonFilters(filters, st)
} else {
  new NoopFilters()
}
```

Reply:
I believe the flag check should be incapsulated in JsonFilters, and not visible at the caller side.

Reply:
👌  that's fine